lib/spack/spack/cmd/buildcache.py

```diff
@@ -338,10 +338,14 @@ def _createtarball(env, spec_yaml, packages, add_spec, add_deps,
     elif packages:
         packages = packages
 
+    elif env:
+        packages = env.roots()
```

Comment:
This should use `env.concretized_user_specs` so that it cannot attempt to operate on an abstract spec (`env.roots` returns concrete specs /for specs that have been concretized/ but otherwise returns the abstract specs).

@shahzebsiddiqui's example is a good one. What if I install `m4` in the environment and call `spack buildcache create`? I would expect to not only get `m4`, but also `libsigsegv` since `m4` depends on `libsigsegv`. Can you confirm that this will add the dependencies in the default case (no `--only` argument)?

Reply:
Thanks @becker33! Will make the change as suggested. 

I confirm that by default the env roots and all their dependencies are added to the build cache. So in this example both `m4` and `libsigsegv` are added.